library/common/jni/jni_interface.cc

```diff
@@ -236,20 +324,28 @@ static void* jvm_on_response_data(envoy_data data, bool end_stream, void* contex
 
 static envoy_filter_data_status jvm_http_filter_on_request_data(envoy_data data, bool end_stream,
                                                                 const void* context) {
-  envoy_data return_data = copy_envoy_data(data.length, data.bytes);
-  jobject result = static_cast<jobject>(
+  JNIEnv* env = get_env();
+  jobjectArray result = static_cast<jobjectArray>(
```

Comment:
Previously we ignored the `result`; now we extract its elements, map, and return them.